src/chainstate/stacks/boot/contract_tests.rs

```diff
@@ -207,6 +250,60 @@ fn test_sim_hash_to_height(in_bytes: &[u8; 32]) -> Option<u64> {
     }
 }
 
+impl BurnStateDB for TestSimBurnStateDB {
+    fn get_burn_block_height(
+        &self,
+        sortition_id: &crate::types::chainstate::SortitionId,
+    ) -> Option<u32> {
+        panic!("Not implemented in TestSim");
+    }
+
+    fn get_burn_header_hash(
+        &self,
+        height: u32,
+        sortition_id: &crate::types::chainstate::SortitionId,
+    ) -> Option<BurnchainHeaderHash> {
+        panic!("Not implemented in TestSim");
+    }
+
+    fn get_stacks_epoch(&self, height: u32) -> Option<StacksEpoch> {
+        let epoch_begin_index = match self.epoch_bounds.binary_search(&(height as u64)) {
```

Comment:
In the structs that use `epoch_bounds`, I think it would be useful to leave a note about how it should be constructed (for example, don't include 0 for the start of Epoch10).

Reply:
Added in https://github.com/blockstack/stacks-blockchain/pull/2659/commits/1b38084371d2788e878b8c2785d7a501544a9333